### PR TITLE
add shadowroot guardtower

### PR DESCRIPTION
**Pull request by pentantan**

### CODE REVIEW

File: src/main/java/com/mushroom/midnight/common/world/feature/structure/ShadowrootGuardTowerPieces.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import net.minecraft.world.gen.feature.template.TemplateManager;
+import net.minecraft.world.storage.loot.LootTables;
+
+import java.util.List;
+import java.util.Random;
+
+public class ShadowrootGuardTowerPieces {
+    private static final ResourceLocation field_202592_e = new ResourceLocation(Midnight.MODID, "shadowroot_guardtower");
+
+    public static void func_207617_a(TemplateManager p_207617_0_, BlockPos p_207617_1_, Rotation p_207617_2_, List<StructurePiece> p_207617_3_, Random p_207617_4_) {
```
Comment (gegy1000) [inline src/main/java/com/mushroom/midnight/common/world/feature/structure/ShadowrootGuardTowerPieces.java]: Hate to nitpick on such small things, but would you be able to rename these unnamed things? 🙂

File: src/main/java/com/mushroom/midnight/common/registry/MidnightFeatures.java [hunk 2]
```diff
@@ -190,11 +197,15 @@ public static void registerFeatures(IForgeRegistry<Feature<?>> event) {
                 .add("crystal_cluster", CRYSTAL_CLUSTER)
                 .add("crystal_spire", CRYSTAL_SPIRE)
                 .add("ore", ORE)
-                .add("well", WELL);
+                .add("well", WELL)
+                .add("shadowroot_guardtower", SHADOWROOT_GUARDTOWER);
+
+        Feature.STRUCTURES.put("ShadowrootGuardTower".toLowerCase(Locale.ROOT), SHADOWROOT_GUARDTOWER);
```
Comment (gegy1000) [inline src/main/java/com/mushroom/midnight/common/registry/MidnightFeatures.java]: Maybe could make a special `RegUtil` helper thing for structures? Might make things easier

**Comment by pentantan** [comment]
I add Structure register utils and fixed some progrem

File: src/main/java/com/mushroom/midnight/common/biome/MidnightBiomeConfigurator.java [hunk 3]
```diff
@@ -358,6 +358,13 @@ public static void addWell(ConfigurableBiome biome) {
         );
     }
 
+    public static void addShadowRootGuardTower(ConfigurableBiome biome) {
+        biome.add(GenerationStage.Decoration.SURFACE_STRUCTURES, Biome.createDecoratedFeature(
```
Comment (gegy1000) [inline src/main/java/com/mushroom/midnight/common/biome/MidnightBiomeConfigurator.java]: I think structures only needed to be added with the structure `add`

Comment (pentantan) [reply src/main/java/com/mushroom/midnight/common/biome/MidnightBiomeConfigurator.java]: To tell the truth, add Structure is not created just by registering a structure.
If you create a structure, you must also run addFeature
If you look at the addStructures of Vanilla's DefaultBiomeFeature, you can see that the structure is added with addFeature

Comment (gegy1000) [reply src/main/java/com/mushroom/midnight/common/biome/MidnightBiomeConfigurator.java]: Odd, villages seem to be added with just addStructure. I wonder what the difference is

Comment (gegy1000) [reply src/main/java/com/mushroom/midnight/common/biome/MidnightBiomeConfigurator.java]: Ah! Nevermind, I see. Vanilla's just all over the place. Sorry :P

Comment (pentantan) [reply src/main/java/com/mushroom/midnight/common/biome/MidnightBiomeConfigurator.java]: no problem :)

**Pull request merged**